SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp

```diff
@@ -60,3 +68,156 @@ double USpatialMetrics::CalculateLoad() const
 
 	return AverageFrameTime / TargetFrameTime;
 }
+
+void USpatialMetrics::SpatialStartRPCMetrics()
+{
+	if (bRPCTrackingEnabled)
+	{
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Already recording RPC metrics"));
+		return;
+	}
+
+	UE_LOG(LogSpatialMetrics, Log, TEXT("Recording RPC metrics"));
+
+	bRPCTrackingEnabled = true;
+	RPCTrackingStartTime = FPlatformTime::Seconds();
+
+	// If RPC tracking is activated on a client, send a command to the server to start tracking.
+	if (!NetDriver->IsServer())
+	{
+		FUnrealObjectRef PCObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromObject(Cast<APlayerController>(NetDriver->GetSpatialOSNetConnection()->OwningActor));
+		Worker_EntityId ControllerEntityId = PCObjectRef.Entity;
+
+		if (ControllerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+		{
+			Worker_CommandRequest Request = {};
+			Request.component_id = SpatialConstants::DEBUG_METRICS_COMPONENT_ID;
+			Request.schema_type = Schema_CreateCommandRequest(SpatialConstants::DEBUG_METRICS_COMPONENT_ID, SpatialConstants::DEBUG_METRICS_START_RPC_METRICS_ID);
+			NetDriver->Connection->SendCommandRequest(ControllerEntityId, &Request, SpatialConstants::DEBUG_METRICS_START_RPC_METRICS_ID);
+		}
+		else
+		{
+			UE_LOG(LogSpatialMetrics, Warning, TEXT("SpatialStartRPCMetrics: Could not resolve local PlayerController entity!"));
```

Comment:
If this happened should we reset bRPCTrackingEnabled and RPCTrackingStartTime?

Reply:
I'm not sure if failing to send the request to the server should invalidate tracking this locally as well.